sycl/test/basic_tests/event_release.cpp

```diff
@@ -0,0 +1,36 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: env SYCL_PI_TRACE=1 %CPU_RUN_PLACEHOLDER %t.out 2>&1 %CPU_CHECK_PLACEHOLDER
+#include <CL/sycl.hpp>
+#include <cassert>
+
+// The test checks that pi_events are released without queue destruction
+// or call to queue::wait, when the corresponding commands are cleaned up.
+
+using namespace cl::sycl;
+
+class Foo;
+
+int main() {
+  int Val = 0;
+  int Gold = 42;
+
+  // Check event releasing without queue destruction
+  queue *QPtr = new queue{};
+
+  {
+    buffer<int, 1> Buf{&Val, range<1>(1)};
+    QPtr->submit([&](handler &Cgh) {
+      auto Acc = Buf.get_access<access::mode::discard_write>(Cgh);
+      Cgh.single_task<Foo>([=]() {
+        Acc[0] = Gold;
+      });
+    });
+  }
+
+  // Buffer destruction triggers execution graph cleanup, check that both
+  // events (one for launching the kernel and one for memory transfer to host)
+  // are released.
+  // CHECK: piEventRelease
+  // CHECK: piEventRelease
```

Comment:
These are checked after application terminated, aren't these?

Reply:
Correct. Anything wrong with that?

Reply:
How do you check that event release occurs earlier than queue's destruction or call to wait.
I suggest adding some output as a marker that the app is going to terminate in a moment. Then you'll be able to check that there are two `piEventRelease`s and that the marker is at rear end of these releases.

Reply:
The queue was neither destroyed nor waited for in the test in the first place, but I actually like this approach better rather than leaving undeleted objects in the test source code, so I applied it anyway.